### PR TITLE
[Feature] Support function next_day

### LINKED ISSUE #19553: [function]next_day/previous_day need to be supported.
## Feature request

it is necessary for user to check next mon/tue/wed... day. here is my proposal:

### description

Returns the date of the first specified DOW (day of week) that occurs after the input date.

### syntax

```
NEXT_DAY( <date_or_datetime_expr> , <dow_string> )
```
in which, 
`date_or_datetime_expr` can be date or datetime type
`dow_string` specifies the day of week.
Validvalues include a number of abbreviations:
Sunday, Su, Sun
Monday, Mo, Mon
Tuesday, Tu, Tue
Wednesday, We, Wed
Thursday, Th, Thu
Friday,  Fr, Fri
Saturday,  Sa, Sat

and return a date.

### PR DESCRIPTION
**Pull request by mlimwxxnn**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #19553

## Problem Summary(Required) ：

NEW FUNCTION: `next_day`

Test sql:
```sql
CREATE TABLE IF NOT EXISTS `next_day_test` (
  `d1` DATE,
  `d2` DATETIME
) 
DISTRIBUTED BY HASH(`d1`) 
PROPERTIES(
  "replication_num" = "1"
);

INSERT INTO `next_day_test`(d1, d2)
VALUES ('2023-04-01', '2023-04-01 20:13:14'),
       ('2023-04-02', '2023-04-02 20:13:14'),
       ('2023-04-03', '2023-04-03 20:13:14'),
       ('2023-04-04', '2023-04-04 20:13:14'),
       ('2023-04-05', '2023-04-05 20:13:14'),
       ('2023-04-06', '2023-04-06 20:13:14'),
       ('2023-04-07', '2023-04-07 20:13:14');

SELECT next_day('2023-04-05', 'Mo');
SELECT next_day('2023-04-05', 'Tue');
SELECT next_day('2023-04-05', 'Wednesday');
SELECT next_day('2023-04-05 20:13:14', 'Thursday');


SELECT d1, next_day(d1, 'Fr'), d2, next_day(d2, 'Fr') FROM next_day_test;
SELECT d1, next_day(d1, 'Sat'), d2, next_day(d2,'Sat') FROM next_day_test;
SELECT d1, next_day(d1, 'Sunday'), d2, next_day(d2, 'Sunday') FROM next_day_test;
```

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctionsTest.java [hunk 1]
```diff
@@ -551,6 +551,46 @@ public void curDate() {
         assertEquals(now, ScalarOperatorFunctions.curDate().getDate());
     }
 
+
+    @Test
+    public void nextDay() {
```
Comment (imay) [inline fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctionsTest.java:556]: How about writing tests with sql-test which is end to end and easy to write

Comment (mlimwxxnn) [reply fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctionsTest.java:556]: Yes, I can rewrite the tests.
For `sql-test`, do you mean the `SQL-tester` framework in `starrocks/test/` path?

Comment (imay) [reply fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctionsTest.java:556]: Yes

**Comment by imay** [comment]
you also need to sign off this PR by `git commit -s`

File: fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctionsTest.java (hunk 1 above)
Comment (mlimwxxnn) [reply fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctionsTest.java:556]: The `SQL-tester` framework is really convenient! I have added the tests and passed.

**Comment by mlimwxxnn** [comment]
I have commit with sign but DCO check also failed.

**Comment by imay** [comment]
1. you need to run `build-support/clang-format.sh` to format the cpp source files.
2. If the last commit has sign-off, DCO will be happy then.

File: test/sql/test_time_fn/T/test_time_fn [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+       ('2023-04-07', '2023-04-07 20:13:14');
+
+
+SELECT next_day('2023-04-05', 'Mo');
+SELECT next_day('2023-04-05', 'Tue');
+SELECT next_day('2023-04-05', 'Wednesday');
+SELECT next_day('2023-04-05 20:13:14', 'Thursday');
+SELECT d1, next_day(d1, 'Fr'), d2, next_day(d2, 'Fr') FROM next_day_test order by d1;
+SELECT d1, next_day(d1, 'Sat'), d2, next_day(d2,'Sat') FROM next_day_test order by d1;
+SELECT d1, next_day(d1, 'Sunday'), d2, next_day(d2, 'Sunday') FROM next_day_test order by d1;
```
Comment (fzhedu) [inline test/sql/test_time_fn/T/test_time_fn:27]: please add more tests:
1. abnormal tests, for example the data fromat or dow format is not right;
2. some special tests, like 2/29 or 2/28 in LEAP year or not.

**Review by fzhedu — COMMENTED** [review]
pleas add doc

File: test/sql/test_time_fn/T/test_time_fn (hunk 2 above)
Comment (fzhedu) [reply test/sql/test_time_fn/T/test_time_fn:27]: 3. inputs with null

Comment (ZiheLiu) [reply test/sql/test_time_fn/T/test_time_fn:27]: And add some test cases, where the second parameter `DOW` is from table not a constant value.

Comment (mlimwxxnn) [reply test/sql/test_time_fn/T/test_time_fn]: 1.abnormal tests
2.special tests (LEAP year, LEAP month)
3.null
4.`DOW` is from table (date and dow from same/different table)
I have added the above cases to the test.

**Pull request closed without merging**